src/plugins/discover/public/application/main/components/layout/discover_main_content.tsx

```diff
@@ -78,6 +78,17 @@ export const DiscoverMainContent = ({
           trackUiMetric(METRIC_TYPE.CLICK, DOCUMENTS_VIEW_CLICK);
         }
       }
+
+      return new Promise<VIEW_MODE>((resolve, reject) => {
```

Comment:
This is used by the Patterns tab to know when it can apply the filters after switching to the Documents tab.
Without waiting for the switch, the filters are not applied.